### PR TITLE
Cancellable crons

### PR DESCRIPTION
**Pull request by Crell**
DEVREL-721

Cancellable cron syntax.

### CODE REVIEW

File: docs/src/configuration/app/cron.md [hunk 1]
```diff
@@ -23,6 +24,44 @@ If an application defines both a `web` instance and a `worker` instance, cron ta
...
+
+The full version is the following:
+
+```yaml
+crons:
+    mycommand:
+        spec: '*/17 * * * *'
+        commands:
+            start: <command here>
+            stop: <command here>
```
Comment (ralt) [inline docs/src/configuration/app/cron.md:39]: I feel like this version is not making it obvious that this field and the shutdown_timeout fields are optional

Comment (chadwcarlson) [reply docs/src/configuration/app/cron.md:36]: @Crell I’m thinking either

a) 
```
crons:
    mycommand:
        spec: '*/17 * * * *'
        commands:
            start: <required; command here>
            stop: <command here>
            shutdown_timeout: <integer here, default 10s>
```
b) or splitting the next paragraphs:

* **`start` (Required):** The `start` key is the cron command that should execute according to the specified schedule.  It is required, and is often all you need.
* **`stop` (Optional):** If a cron task is interrupted by a user through the CLI or Web Admin Console, then the `stop` command will be issued to give the cron command a chance to shutdown gracefully, such as finish an active item in a list of tasks.  The `stop` command is optional, and if not specified then a `SIGTERM` signal will be sent to the process.
* **`shutdown_timeout` (Optional):** If the cron process is still running after `shutdown_timeout` seconds, a `SIGKILL` signal will be sent to the process to force terminate it.  If not specified, the default `shutdown_timeout` is 10 seconds.

b is probably more clear

Comment (Crell) [reply docs/src/configuration/app/cron.md:36]: I went with B, using the formatting convention already in place on the main app.yaml page.

File: docs/src/configuration/app/upgrading.md [hunk 2]
```diff
@@ -3,6 +3,30 @@ title: "Upgrading"
...
+```yaml
+crons:
+    sendemails:
+        spec: '*/7 * * * *'
+        commands:
+            start: cd public && send-pending-emails.sh
+            shutdown_timeout: 10
+```
+
+The new syntax offers greater flexibility and configuration.  See the [cron documentation]({{< relref "/configuration/app/cron.md" >}}) for more details.
```
Comment (chadwcarlson) [inline docs/src/configuration/app/upgrading.md:28]:
```suggestion
The new syntax offers greater flexibility and configuration.  See the [cron documentation](/configuration/app/cron.md) for more details.
```

**Comment by CollierCZ** [comment]
Closing for now. Will reopen or recreate when the feature is closer to done.

**Comment by CollierCZ** [comment]
Will be done in #2141

**Pull request closed without merging**